packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import uniqueid from 'lodash.uniqueid';
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+    * Whether or not the Switch is enabled ("ON").
+    */
+  isOn: PropTypes.bool,
+
+  /**
+    * If true, the user won't be able to toggle the switch.
+    */
+  isDisabled: PropTypes.bool,
+
+  /**
+    * Text of the label.
+    */
+  labelText: PropTypes.node.isRequired,
+
+  /**
+    * Callback fired when the state is changed. Parameters are (event).
+    */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  isOn: false,
+  isDisabled: false,
+  onChange: undefined,
+};
+
+const Switch = ({
+  isOn,
+  isDisabled,
+  onChange,
+  labelText,
+  ...customProps
+}) => {
+  const [focused, setFocus] = useState(false);
+  const sliderButtonRef = useRef(null);
+
+  const handleClick = (event) => {
+    setFocus(true);
+    sliderButtonRef.current.focus();
+    onChange(event);
+  };
+
+  const handleOnKeyDown = (event) => {
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      event.preventDefault();
+      setFocus(true);
+    }
+  };
+
+  const handleOnBlur = () => {
+    setFocus(false);
+  };
+
+  const handleKeyUp = (event) => {
+    if (event.keyCode === KeyCode.KEY_TAB) {
+      setFocus(true);
+    } else if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      setFocus(true);
+      onChange(event);
```

Comment:
How can the user established the checked state from the event? Is there a way to access a value? An input for example would return an event in the onChange that allowed consumers to access the value via event.target.value.

Looking into this more I think `onChange` should return a boolean instead of an event unless there is a way to extract the checked state from the event.

Reply:
Please correct me if my understanding is wrong . IMO since we have `isOn` prop to set the switch on and off why would user need to be dependent on return value of OnChange to get the checked (current) state of switch. when user can refer to the state or value which is used to set `isOn` prop. because switch state is completely dependent on the `isOn` prop value so I think state or value variable will always be aligned with switch state.

Reply:
The responsibility of an onChange function is to return the changed state of a component regardless of the complexity. The consuming application can decide whether or not to use that value. What is the purpose of returning the event if it cannot be used to interpret the changed value?

Reply:
Makes sense returning boolean value with [Onchange](https://github.com/cerner/terra-core/blob/895c4fad7bb180e27e1919296ac90eeaa805dea1/packages/terra-switch/src/Switch.jsx#L63) callBack. 
Event parameter can be used to get the event type if it required to do any events specific functions.